src/app/profile/yourGiving/stopStartRecurringGifts/restartGift/step3/confirmGifts/confirmGifts.component.js

```diff
@@ -0,0 +1,75 @@
+import angular from 'angular';
+import template from './confirmGifts.tpl';
+import isEmpty from 'lodash/isEmpty';
+import concat from 'lodash/concat';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/observable/forkJoin';
+import 'rxjs/add/operator/do';
+
+import giftListItem from 'common/components/giftViews/giftListItem/giftListItem.component';
+import giftDetailsView from 'common/components/giftViews/giftDetailsView/giftDetailsView.component';
+
+import donationsService from 'common/services/api/donations.service';
+
+let componentName = 'confirmGifts';
+
+class ConfirmGiftsController {
+
+  /* @ngInject */
+  constructor( donationsService ) {
+    this.donationsService = donationsService;
+  }
+
+  $onInit() {
+    this.updates = [];
+    this.adds = [];
+    this.saved = [];
+    angular.forEach( this.gifts, ( gift ) => {
+      (angular.isDefined( gift.parentDonation['donation-status'] ) ? this.updates : this.adds).push( gift );
+    } );
+  }
+
+  processRestarts() {
+    let requests = [];
+    this.setLoading( {loading: true} );
+    if ( !isEmpty( this.adds ) ) {
+      requests.push( this.donationsService.addRecurringGifts( this.adds ).do( () => {
+        this.saved = concat( this.saved, this.adds );
+        this.adds.length = 0;
```

Comment:
I tested this in the specs, if either one succeeds and the other fails, it moves saved items to the saved array and sets an error state. The UI changes the "Back" button to a "Cancel" in this case and doesn't allow the user to go back. If we need to allow it, I'll need to do some additional work to un-select saved gifts. I clone the gifts before sending them to this component so that changes made to them aren't changing the originals.